File: 03_time_complexity/tape_equilibrium.py
def solution(A):
    pdiff = A[0]-sum(A[1:])
    result = abs(pdiff)
    
    l = len(A)
    for i in xrange(1, l-1):
        pdiff += 2 * A[i]
        result = min(result, abs(pdiff))
    return result

File: 03_time_complexity/tape_equilibrium.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> &a) {
  int n = a.size();
  int s = 0;
  int ret;
  for (int i = 1; i < n; i++)
    s += a[i];
  s -= a[0];
  ret = abs(s);
  for (int i = 1; i < n-1; i++) {
    s -= 2 * a[i];
    ret = min(ret, abs(s));
  }
  return ret;
}
